src/pudl/output/pudltabl.py

```diff
@@ -1127,32 +1146,64 @@ def gens_mega_eia(self, update: bool = False) -> pd.DataFrame:
                     "The frequency of the pudl_out object must be `AS` for the "
                     f"plant-parts table and we got {self.freq}"
                 )
+            if gens_cols is None:
+                gens_cols = [
+                    "technology_description",
+                    "energy_source_code_1",
+                    "prime_mover_code",
+                    "operating_date",
+                    "retirement_date",
+                    "operational_status",
+                    "capacity_mw",
+                ]
             self._dfs[
                 "gens_mega_eia"
             ] = pudl.analysis.plant_parts_eia.MakeMegaGenTbl().execute(
-                mcoe=self.mcoe(all_gens=True), own_eia860=self.own_eia860()
+                mcoe=self.mcoe(all_gens=True, gens_cols=gens_cols),
+                own_eia860=self.own_eia860(),
             )
         return self._dfs["gens_mega_eia"]
 
     def plant_parts_eia(
         self,
         update: bool = False,
         update_gens_mega: bool = False,
+        gens_cols: Any = None,
     ) -> pd.DataFrame:
         """Generate and return master plant-parts EIA.
 
         Args:
             update: If true, re-calculate the output dataframe, even
                 if a cached version exists.
             update_gens_mega: If True, update the gigantic Gens Mega table.
+            gens_cols: equal to the string "all", None, or a list of
+                additional column attributes to include from the EIA 860 generators table
+                in the output mega gens table. By default all columns necessary to create
+                the EIA plant part list are included.
         """
         update_any = any([update, update_gens_mega])
         if update_any or self._dfs["plant_parts_eia"] is None:
+            # default columns needed to create plant part list
+            if gens_cols is None:
+                gens_cols = [
+                    "technology_description",
+                    "energy_source_code_1",
+                    "prime_mover_code",
+                    "operating_date",
+                    "retirement_date",
+                    "operational_status",
+                    "capacity_mw",
+                    "fuel_type_code_pudl",
+                    "planned_retirement_date",
```

Comment:
why are these last two columns different than the columns in `gens_mega_eia`? because these two are needed for the full ppl but aren't for the gens_mega?

Reply:
yesss exactly